packages/loot-core/src/server/rules/types/handlers.ts

```diff
@@ -18,13 +18,13 @@ export interface RulesHandlers {
 
   'rule-add': (
     rule: Omit<RuleEntity, 'id'>,
-  ) => Promise<{ error: ValidationError } | { id: string }>;
+  ) => Promise<{ error: ValidationError } | RuleEntity>;
 
   'rule-update': (
-    rule: Partial<RuleEntity>,
```

Comment:
<!-- Warning represent user-facing problems. If possible, it's great to follow this kind of comment with a suggestion.  -->
**🔨 warning:** ‏I think `Partial<RuleEntity>` was correct here. You don't need to pass a full rule object here in order to perform updates. Only pass the fields you want to get updated.

Reply:
I see, thanks for clarifying. I guess the return type needs to be a `Partial` then

Reply:
Yea, I think so.